pkg/jobs/registry.go

```diff
@@ -363,11 +364,28 @@ func (r *Registry) maybeCancelJobs(ctx context.Context, nl NodeLiveness) {
 	}
 }
 
+// isOrphaned tries to detect if there are no mutations left to be done for the
+// job which will make it a candidate for garbage collection. Jobs can be left
+// in such inconsistent state if they fail before being removed from the jobs table.
+func (r *Registry) isOrphaned(ctx context.Context, payload *jobspb.Payload) (bool, error) {
+	if payload.Type() != jobspb.TypeSchemaChange {
+		return false, nil
+	}
+	for _, id := range payload.DescriptorIDs {
+		txn := r.db.NewTxn(ctx, "get_table_and_db_descrs")
```

Comment:
we usually use `r.db.Txn()` and pass a closure, which also means you don't need to do your own retries.